portal/components/table/_hooks/useTableData.tsx

```diff
@@ -0,0 +1,55 @@
+import { useMemo } from 'react'
+import Skeleton from 'react-loading-skeleton'
+
+import { TableProps } from '../index'
+
+type Props<T> = Omit<
+  TableProps<T>,
+  'data' | 'loading' | 'skeletonRows' | 'smallBreakpoint'
+> &
+  Required<
+    Pick<TableProps<T>, 'loading' | 'skeletonRows' | 'smallBreakpoint'>
+  > & {
+    data: T[] | undefined
+    width: number
+  }
```

Comment:
IF we apply [this comment](https://github.com/hemilabs/ui-monorepo/pull/1520/files#r2356340616), we could remove this as it would be picked from TableProps

```suggestion
  Required<
    Pick<TableProps<T>, 'data' | 'loading' | 'skeletonRows' | 'smallBreakpoint'>
  > & {
    width: number
  }
```